### ISSUE TITLE
Publishing to the credential registry staging server fails without an error message.

### ISSUE THREAD

**Issue by cwd-mparsons** [CredentialEngine/CredentialRegistry#230]
@rsaksida @edgarf 
We had set up our staging environments some time ago. I have been updating the environments and doing test publishing. 

We had previously set up the registry authorization token of :	9aa4f8174d882721661fa5a702b7a75a
The latter no longer works – get a 401 error.

Since the registry production data was copied to the staging registry, I tried the production key of: b8b3c833eea…….
It appeared to work in that I didn’t get a 401 error, but a success status code of false was returned, but no message:
{"errors":[""]}

**Comment by cwd-mparsons** [comment]
@rsaksida @edgarf 
I added some tracing to capture the full HttpResponseMessage. The only additional information is that there was an Internal Server Error:

{
	"Version":{	"Major":1,"Minor":1,"Build":-1,"Revision":-1,"MajorRevision":-1,"MinorRevision":-1},
	"Content":{	"Headers":[{"Key":"Content-Length","Value":["15"]},{"Key":"Content-Type","Value":["application/json"]}]},
	"StatusCode":**500**,
	"ReasonPhrase":"**Internal Server Error"**,
	"Headers":[
		{"Key":"Connection","Value":["keep-alive"]},
		{"Key":"Status","Value":["500 Internal Server Error"]},
		{"Key":"Vary","Value":["Origin"]},
		{"Key":"Date","Value":["Tue, 11 Jun 2019 19:51:54 GMT"]},
		{"Key":"X-Powered-By","Value":["Phusion Passenger 5.1.11"]},
		{"Key":"Server",
		"Value":["nginx","+","Phusion","Passenger","5.1.11"]}
	],
	"RequestMessage":{
	"Version":{
		"Major":1,"Minor":1,"Build":-1,"Revision":-1,
		"MajorRevision":-1,
		"MinorRevision":-1},
	"Content":{
		"Headers":[
			{"Key":"Content-Type",
				"Value":["application/json; charset=utf-8"]},
			{"Key":"Content-Length","Value":["31110"]}
			]},
	"Method":{
		"Method":"POST"},
	"RequestUri":"https://staging.credentialengineregistry.org/resources/organizations/cbd294be-7b7c-48e7-ab98-dee849e50a68/documents?skip_validation=true",
	"Headers":[
		{"Key":"Authorization","Value":["Token b8b3c833eea91bd338953abf33303241"]}
	],
	"Properties":{}},
	"IsSuccessStatusCode":false
}

**Comment by excelsior** [comment]
@cwd-mparsons Could you provide the request you made (with payload and stuff), so I can investigate why it fails?

**Comment by cwd-mparsons** [comment]
@excelsior Note that there are two publishing methods possible: one where the client manages the public/private keys and the second where the registry manages the keys. 
The error is with the second method. So all types of publishing, credential, organization, etc fail. 
The following gist is for a concept scheme

https://gist.github.com/cwd-mparsons/6dc3d311217d1c287c61ff15c3e339cd#file-audience_conceptscheme-json

**Comment by cwd-mparsons** [comment]
@excelsior I just tried to publish a credential to the staging registry and have the same scenario where the publish fails and no message is returned in the 'errors' property. 
{"errors":[""]}

However the content returned includes a statusCode of 500 and a reason of Internal Server Error.
{"Version":{"Major":1,"Minor":1,"Build":-1,"Revision":-1,"MajorRevision":-1,"MinorRevision":-1},"Content":{"Headers":[{"Key":"Content-Length","Value":["15"]},{"Key":"Content-Type","Value":["application/json"]}]},"**StatusCode":500,"ReasonPhrase":"Internal Server Error"**,"Headers":[{"Key":"Connection","Value":["keep-alive"]},{"Key":"Status","**Value":["500 Internal Server Error"]**},{"Key":"Vary","Value":["Origin"]},{"Key":"Date","Value":["Wed, 21 Aug 2019 01:00:41 GMT"]},{"Key":"X-Powered-By","Value":["Phusion Passenger 5.3.7"]},{"Key":"Server","Value":["nginx","+","Phusion","Passenger","5.3.7"]}],"RequestMessage":{"Version":{"Major":1,"Minor":1,"Build":-1,"Revision":-1,"MajorRevision":-1,"MinorRevision":-1},"Content":{"Headers":[{"Key":"Content-Type","Value":["application/json; charset=utf-8"]},{"Key":"Content-Length","Value":["18211"]}]},"Method":{"Method":"POST"},"RequestUri":"https://staging.credentialengineregistry.org/resources/organizations/ef79b83b-0bf2-4f32-9837-dd3acb1c1a41/documents?skip_validation=true","Headers":[{"Key":"Authorization","Value":["Token b8b3c833eea91bd338953abf33303241"]}],"Properties":{}},"IsSuccessStatusCode":false}

**Comment by excelsior** [comment]
The problem was caused by the wrong private key secret used on staging. We restored the staging DB from a production snapshot awhile ago (when troubleshooting the indexing mechanism), so staging has the key pairs from prod, but the private key secret stayed the same, thus encryption failed during self-signing publishing, yet the other method worked as usual. I changed the secret, now publishing should work fine either way.

Question: Do you still need production data on staging or it should be an independent environment?

**Comment by cwd-mparsons** [comment]
@excelsior No, we don't need production data on staging. It is to be independent from other environments. I had previously cleared out the envelopes. The organizations from production should be removed.

**Comment by excelsior** [comment]
I removed the prod organizations from staging, there are only four organizations left there now: https://staging.credentialengineregistry.org/metadata/organizations

I also restored the previous secret key, now publishing on behalf should work again.
